### PR TITLE
feat: create encoder

### PR DESCRIPTION
**Pull request by tadejsv**

### CODE REVIEW

File: audioclip_text.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            docs,
+            traversal_path=parameters.get(
+                'traversal_paths', self.default_traversal_paths
+            ),
+            batch_size=parameters.get('batch_size', self.default_batch_size),
+            needs_attr='text',
+        )
+
+        for batch in batch_generator:
+            ((_, _, embeddings), _), _ = self.model(text=[[doc.text] for doc in batch])
```
Comment (JoanFM) [inline audioclip_text.py:57]: do not use the forward method, use the encode_text public one

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: tests/integration/test_encoder.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+def test_integration(data_generator: Callable, request_size: int):
+    with Flow(return_results=True).add(uses=AudioCLIPTextEncoder) as flow:
+        resp = flow.post(
+            on="/index", inputs=data_generator(), request_size=request_size, return_results=True
+        )
+
+    assert min(len(resp) * request_size, 50) == 50
+    for r in resp:
+        for doc in r.docs:
+            assert doc.embedding is not None
```
Comment (bwanglzu) [inline tests/integration/test_encoder.py:21]: can you also assert the shape is expected?

**Review by bwanglzu — CHANGES_REQUESTED** [review]
great work! 
for this fixture:
```python
@pytest.fixture()
def docs_with_chunk_chunk_text() -> DocumentArray:
    return DocumentArray(
        [
            Document(
                chunks=[
                    Document(chunks=[Document(text='hello world') for _ in range(10)])
                ]
            )
        ]
    )
```
no need to change, just a short remind it is not a recommended way to set up chunks in Document, remember we lost the `granularity`?

File: audioclip_text.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    ):
+        super().__init__(*args, **kwargs)
+
+        self.model = AudioCLIP(pretrained=model_path).to(device).eval()
+        self.default_traversal_paths = default_traversal_paths
+        self.default_batch_size = default_batch_size
+
+    @requests
+    def encode(
+        self, docs: Optional[DocumentArray], parameters: dict, *args, **kwargs
```
[the remark this answers is not in the document]
Comment (tadejsv) [reply audioclip_text.py:43]: According to [what I was told](https://github.com/jina-ai/executor-text-clip-text-encoder/pull/9#discussion_r674564807), the default is to give `{}` if no parameters are passed. So then for each default value you can do `'default_value' in parameters` to check for it. I think this is better than allowing `parameters` to be `None`, as you don't need to do an additional check to see if this argument was passed at all.

Also, all the other encoders also have this set up like this, so I would keep it like this for consistency.

File: tests/integration/test_encoder.py (hunk 2 above)
Comment (tadejsv) [reply tests/integration/test_encoder.py]: Done

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Review by davidbp — APPROVED** [review]
LGTM!

File: audioclip_text.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            traversal_path=parameters.get(
+                'traversal_paths', self.default_traversal_paths
+            ),
+            batch_size=parameters.get('batch_size', self.default_batch_size),
+            needs_attr='text',
+        )
+
+        with torch.no_grad():
+            for batch in batch_generator:
+                embeddings = self.model.encode_text(text=[[doc.text] for doc in batch])
```
Comment (JoanFM) [inline audioclip_text.py:66]: u shouls use batch.get_attributes

File: audioclip_text.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                embeddings = embeddings.cpu().numpy()
+
+                for idx, doc in enumerate(batch):
```
Comment (JoanFM) [inline audioclip_text.py:69]: here we should use zip like iteration

**Pull request merged**